specs/order_spec.rb

```diff
@@ -110,22 +110,45 @@
       # The list of products should not have been modified
       expect(order.total).must_equal before_total
     end
+
+    # optional test
+    it "Raises an ArgumentError if the product was not found" do
+      # Arrange
+      products = { "banana" => 1.99, "cracker" => 3.00 }
```

Comment:
You've put this test in the `describe` block for `add_product`, but this test is for `remove_product`. You should create a separate `describe` block for this.

You're also missing _nominal_ cases for `remove_product`. What happens if you try to remove a product that is in the cart?